src/engine/Cooldown.java

```diff
@@ -51,13 +53,41 @@ protected Cooldown(final int milliseconds, final int variance) {
 	 * 
 	 * @return Cooldown state.
 	 */
+
 	public final boolean checkFinished() {
 		if ((this.time == 0)
 				|| this.time + this.duration < System.currentTimeMillis())
 			return true;
 		return false;
 	}
 
+	/**
+	 * Checks if the cooldown has as much time left as 'alert'.
+	 *
+	 * @return Alert state.
+	 */
+
+	public final boolean checkAlert() {
+		if ((this.time > 0)
+				&& this.time + this.duration - this.alert <= System.currentTimeMillis())
+			return true;
+		return false;
+	}
+ 
+  // 경고등 애니메이션 //
```

Comment:
I think, we should write comment by english.
If you want to use korean, please attach english comment too.